js/src/hooks/useAdsSetupCompleteCallback.js

```diff
@@ -34,21 +34,29 @@ export default function useAdsSetupCompleteCallback() {
 	}, [ createNotice ] );
 
 	const handleFinishSetup = useCallback(
-		(
+		async (
 			amount,
 			countryCodes,
 			hasConfirmedEuPoliticalContent,
 			onCompleted
 		) => {
 			setLoading( true );
-			return createAdsCampaign(
-				amount,
-				countryCodes,
-				hasConfirmedEuPoliticalContent
-			)
-				.then( completeAdsSetup )
-				.then( onCompleted )
-				.catch( () => setLoading( false ) );
+
+			try {
+				const payload = await createAdsCampaign(
+					amount,
+					countryCodes,
+					hasConfirmedEuPoliticalContent
+				);
+
+				await completeAdsSetup();
+				await onCompleted?.( payload );
+
+				return payload;
+			} catch ( error ) {
+				setLoading( false );
```

Comment:
Loading is set to `false` only when error occurs, do we need to set in success path as well?